pkg/apis/serving/v1alpha2/kfservice_defaults.go

```diff
@@ -20,7 +20,13 @@ package v1alpha2
 func (kfsvc *KFService) Default() {
 	logger.Info("Defaulting KFService", "namespace", kfsvc.Namespace, "name", kfsvc.Name)
 	kfsvc.Spec.Default.Predictor.ApplyDefaults()
+	if kfsvc.Spec.Default.Explainer != nil {
```

Comment:
Worth abstracting this?

We're taking an EndpointSpec and applying defaults to both predictor and explainer. We need to do that for both default and canary.